src/Mvc/Mvc.Core/src/PublicAPI.Shipped.txt

```diff
@@ -617,8 +617,6 @@ Microsoft.AspNetCore.Mvc.ModelBinding.Validation.ValidateNeverAttribute
 Microsoft.AspNetCore.Mvc.ModelBinding.Validation.ValidateNeverAttribute.ShouldValidateEntry(Microsoft.AspNetCore.Mvc.ModelBinding.Validation.ValidationEntry entry, Microsoft.AspNetCore.Mvc.ModelBinding.Validation.ValidationEntry parentEntry) -> bool
 Microsoft.AspNetCore.Mvc.ModelBinding.Validation.ValidateNeverAttribute.ValidateNeverAttribute() -> void
 Microsoft.AspNetCore.Mvc.ModelBinding.Validation.ValidationVisitor
-Microsoft.AspNetCore.Mvc.ModelBinding.Validation.ValidationVisitor.AllowShortCircuitingValidationWhenNoValidatorsArePresent.get -> bool
```

Comment:
@davidfowl I guess we need both these files 👍

Reply:
I think the "right" way to do this, is to add a line with  `*REMOVED* <rest of API sig>` to the Unshipped file, and then merge them when shipping.

But having something watch that we don't make accidental changes to the Shipped files would be smart.

See implementation [here](https://github.com/dotnet/roslyn-analyzers/blob/c26b04af4f237e17fc50053339d6641c9df67b29/src/PublicApiAnalyzers/Core/Analyzers/DeclarePublicApiAnalyzer.cs#L270-L275)

Reply:
I tried this on a different PR and I see this:

error RS0024: The contents of the public API files are invalid: The shipped API file can't have removed members

@mavasani what's the right way to approach this?

Reply:
@pranavkm The Removed prefix is only meant for unshipped file. When the unshipped file is merged back into the shipped file, the net effect is to just remove the removed API entry completely from shipped file. @333fred wrote a nice script to automate merging unshipped file into shipped file, which we use for Roslyn: https://github.com/dotnet/roslyn/blob/e782801c14884215d4abb9271ad0b360b0def592/scripts/PublicApi/mark-shipped.ps1#L14-L33

Reply:
(I take no credit for that script, I believe that honor goes to @tmat).